x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -105,12 +105,6 @@ export class ActionExecutor {
       throw new Error('ActionExecutor not initialized');
     }
 
-    if (!this.isESOCanEncrypt) {
```

Comment:
Moving this check so we only check if `isESOCanEncrypt` if we are using connector SOs (which we need to decrypt). If this action is using preconfigured connectors, we will not throw if `isESOCanEncrypt` is false